backend/src/main/scala/com/softwaremill/adopttapir/starter/StarterService.scala

```diff
@@ -2,17 +2,60 @@ package com.softwaremill.adopttapir.starter
 
 import cats.effect.IO
 import com.softwaremill.adopttapir.logging.FLogging
+import com.softwaremill.adopttapir.template.{FileTemplate, ProjectTemplate}
 import com.softwaremill.adopttapir.util.Clock
 
 import java.io.File
+import java.util.zip.Deflater
+import scala.reflect.io.Directory
 
 class StarterService(
     clock: Clock,
-    config: StarterConfig
+    config: StarterConfig,
+    template: ProjectTemplate
 ) extends FLogging {
+
   def generateZipFile(starterDetails: StarterDetails): IO[File] = {
-    IO {
-      new java.io.File("src/main/resources/temporal.zip")
+    val filesToCreate = prepareTemplateFiles(starterDetails)
+    IO(os.temp.dir(prefix = config.tempPrefix).toIO)
+      .bracket { tempDir =>
+        for{
+          _ <- logger.debug("created temp dir: " + tempDir.toString)
+          _ <- storeFiles(tempDir, filesToCreate)
+          // TODO: Storage leak: Zip files needs to be cleaned! In future just add something like IO cron job which will
+          //  periodically cleans the data older than 5 min with specific config.tempPrefix.
+          dir <- zipDirectory(tempDir)
+        } yield dir
+      }(release = tempDir => if (config.deleteTempFolder) deleteRecursively(tempDir) else IO.unit)
+  }
+
+  private def prepareTemplateFiles(starterDetails: StarterDetails) = {
```

Comment:
it's good to add type annotations to methods - makes it easier to read

Reply:
Sure  I will add it